hardware_isolation_manager.cpp

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "hardware_isolation_manager.hpp"
+
+#include "utils.hpp"
+
+#include <fmt/format.h>
+
+#include <phosphor-logging/elog-errors.hpp>
+
+namespace hw_isolation
+{
+
+using namespace phosphor::logging;
+
+Manager::Manager(sdbusplus::bus::bus& bus) : _bus(bus)
+{}
+
+std::optional<uint32_t>
+    Manager::getEID(const sdbusplus::message::object_path& bmcErrorLog) const
+{
+    try
+    {
+        constexpr auto loggingObjectPath = "/xyz/openbmc_project/logging";
+        auto loggingInterface = "org.open_power.Logging.PEL";
+        uint32_t eid;
+
+        auto dbusServiceName = utils::getDBusServiceName(
+            _bus, loggingObjectPath, loggingInterface);
+
+        auto method =
+            _bus.new_method_call(dbusServiceName.c_str(), loggingObjectPath,
+                                 loggingInterface, "GetPELIdFromBMCLogId");
+
+        method.append(static_cast<uint32_t>(std::stoi(bmcErrorLog.filename())));
+        auto resp = _bus.call(method);
+
+        resp.read(eid);
+        return eid;
+    }
+    catch (const sdbusplus::exception::SdBusError& e)
+    {
+        log<level::ERR>(fmt::format("Exception [{}] to get EID (aka PEL ID) "
```

Comment:
IMO an error should be returned and guard record should not be created if an invalid error log is passed

Reply:
Right, here adding trace and at last returning as `return std::nullopt;` so caller of this API can decide whether need to handle or throw an exception. 

One of example: https://github.com/ibm-openbmc/openpower-hw-isolation/pull/2/commits/9b1ee2a92148bdd5995dcd60d4e0ce476fa40816#diff-5263f98e5cdba1f39332153cad7bedef71d58e87b9a39a2681cb297e095438d0R230

Reply:
fine